src/main/java/com/AdServlet.java

```diff
@@ -0,0 +1,45 @@
+package com;
+
+import com.google.gson.Gson;
+
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@WebServlet("/ads/*")
+public class AdServlet extends HttpServlet {
+    private static AdCollection adCollection = AdDataBase.adCollection;
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws IOException, ServletException {
+        String id = req.getParameter("id");
+        resp.setContentType("application/json");
+        Ad ad = adCollection.get(id);
+        resp.getWriter().print((new Gson()).toJson(ad));

```

Comment:
Will probably be a Null Pointer Exception if no item found by id. Consider adding a check for null and then setting the HTTP response status to e.g. 404 - not found